core/src/test/java/org/web3j/tx/ReadonlyTransactionManagerTest.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2019 Web3 Labs LTD.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package org.web3j.tx;
+
+import java.io.IOException;
+
+import org.junit.Test;
+
+import org.web3j.protocol.Web3j;
+import org.web3j.protocol.core.DefaultBlockParameterName;
+import org.web3j.protocol.core.Request;
+import org.web3j.protocol.core.methods.response.EthCall;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class ReadonlyTransactionManagerTest {
+
+    EthCall response = mock(EthCall.class);
+    Request request = mock(Request.class);
+    Web3j web3j = mock(Web3j.class);
```

Comment:
It's probably worth mentioning that everything here is mocked, so as far as I can tell nothing is actually being tested. That said this stuff is hard to test in an atomic way.

Reply:
Agreed. I think the approach should mock the `Web3jService` instance passed to `Web3j.build(...)`

Reply:
Something like:

```
Web3jService service = mock(Web3jService.class);
when(service.send()).thenReturn("test");
Web3j web3j = web3j.build(service);
```

Then you can create the ReadonlyTransactionManager with a concrete instance of web3j.